File: docs/src/testers/quickstart/cdp_session.ts
import { chromium } from 'playwright-core';
import { NotteClient } from 'notte-sdk';

const notte = new NotteClient();

await notte.Session({ idle_timeout_minutes: 2 }).use(async (session) => {
  const status = await session.status();
  const cdpUrl = status.cdp_url;

  if (!cdpUrl) {
    throw new Error('Session did not return a cdp_url');
  }

  const browser = await chromium.connectOverCDP(cdpUrl);

  try {
    const page = browser.contexts()[0].pages()[0];
    await page.goto('https://www.google.com');
    await page.screenshot({ path: process.env.NOTTE_SCREENSHOT_PATH || 'screenshot.png' });
  } finally {
    await browser.close();
  }
});

File: docs/src/testers/quickstart/cdp_session.py
import os

from notte_sdk import NotteClient
from playwright.sync_api import sync_playwright

client = NotteClient()

with client.Session(idle_timeout_minutes=2) as session:
    cdp_url = session.cdp_url()

    with sync_playwright() as p:
        browser = p.chromium.connect_over_cdp(cdp_url)
        try:
            page = browser.contexts[0].pages[0]
            page.goto("https://www.google.com")
            page.screenshot(path=os.environ.get("NOTTE_SCREENSHOT_PATH", "screenshot.png"))
        finally:
            browser.close()
